contrib/systemd/etcd2-backup-coreos/README.md

```diff
@@ -0,0 +1,255 @@
+
+- `etcd2-join.service`
+   A oneshot service which wipes all etcd2 data and re-joins the new cluster. This is for adding members **after** the **founding member** has succesfully established the new cluster via `etcd2-restore.service`
+
+## Recovery
+
+This assumes that your cluster has lost majority, and is not recoverable. Otherwise you should probably try to heal your cluster first.
+
+There are a two factors which contribute to how "close" to up-to-date your backup is.
+
+- `etcd2-backup.timer` is configured by default to run a backup every 30 seconds.
+
+- etcd2 has a parameter `snapshot-count` parameter, which controls how many transactions can be committed before the snapshot is written to disk.
+
+Let's compute the "maximum outdatedness" of a backup
+
+Assumptions
+- transaction rate is a constant `1000 transactions / second`
+- `etcd2-backup.timer` is configured for a 30 second interval
+- `etcd2 snapshot-count=10000`
+
+```
+max-missed-seconds= (10000 transactions / (1000 transactions / second)) + 30 seconds = 40 seconds
+```
+
+
+1. Make sure `etcd2.service` and `etcd2-backup.timer` are stopped on all nodes in the cluster
+
+2. Restore the **founding member** by starting `etcd2-restore.service` and then, if successful, `etcd2.service`
+
+3. Restore the rest of the cluster **one at a time**. Start `etcd2-join.service`, and then, if successful, `etcd2.service`. Please verify with `etcdctl cluster-health` that the expected set of nodes is present and healthy after you join each node.
+
+4. Verify that your data is sane (enough). If so, kick off `etcd2-backup.timer` on all nodes and, hopefully, go back to bed.
+
+## Retroactively change the founding member
+
+This will need to happen if you want to restore a cluster from some node's data other than the original founding member.
+
+Change the value of `ETCD_RESTORE_MASTER_ADV_PEER_URLS` in `30-etcd2-backup-restore.conf` to the advertised peer url of the new master node. Repeat the install process, then proceed with recovery instructions.
+
+
+#Example
+
+Let's pretend that we have an initial 3 node CoreOS cluster that we want to configure to back up to S3.
+
+
+| ETCD_NAME  | ETCD_ADVERTISED_PEER_URL |
+| ------------- |:-------------:|
+| e1   | http://172.17.4.51:2379 |
+| e2   | http://172.17.4.52:2379 |
+| e3   | http://172.17.4.53:2379 |
+
+In the event that the cluster fails, we want to restore from `e1`'s backup
+
+## Configuration
+
+```
+[Service]
+Environment="ETCD_RESTORE_MASTER_ADV_PEER_URLS=http://172.17.4.51:2379"
+Environment="RCLONE_ENDPOINT=s3-testing-conf:s3://etcd2-backup-bucket/backups"
+```
+
+The `./rclone.conf` file should contain a section that looks like this:
+
+```
+[s3-testing-conf]
+type = s3
+access_key_id = xxxxxxxx
+secret_access_key = xxxxxx
+region = us-west-1
+endpoint =
+location_constraint =
+```
+
+## Installation
+
+```sh
+cd etcd2-backup
+./build
+scp etcd2-backup.tgz core@e1:~/
+ssh core@e1
+e1 $>  mkdir -p ~/etcd2-backup
```

Comment:
we tend to use just `$` to distinguish shell prompt. However, I see why you want the hostname in there for this particular section. Suggest dropping the >, so you have just:
`e1 $ mkdir [...]`